### PR TITLE
planner: unite the variable name

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/planner/core/plan.go [hunk 1]
```diff
@@ -67,7 +67,8 @@ type Plan interface {
 	// SetOutputNames sets the outputting name by the given slice.
 	SetOutputNames(names types.NameSlice)
 
-	SelectBlockOffset() int
+	// QBOffset is query block offset.
```
Comment (qw4990) [inline pkg/planner/core/plan.go:70]: How about adding an example:
```
// For example, in query `select /*+ use_index(@sel_2 t2, a) */ * from t1, (select a*2 as b from t2) tx where a>b`, 
// the hint should be applied on the sub-query, whose query block is 2.
```

File: pkg/util/hint/hint_query_block.go [hunk 2]
```diff
@@ -283,14 +290,14 @@ func (p *QBHintHandler) GetCurrentStmtHints(hints []*ast.TableOptimizerHint, cur
...
+		p.QBOffsetToHints[offset] = append(p.QBOffsetToHints[offset], hint)
 	}
-	return p.SelOffsetToHints[currentOffset]
+	return p.QBOffsetToHints[currentOffset]
 }
 
 // GenerateQBName builds QBName from offset.
-func GenerateQBName(nodeType NodeType, blockOffset int) (model.CIStr, error) {
-	if blockOffset == 0 {
+func GenerateQBName(nodeType NodeType, selectOffset int) (model.CIStr, error) {
```
Comment (qw4990) [inline pkg/util/hint/hint_query_block.go:299]: selectOffset --> QBOffset

File: pkg/planner/core/initialize.go [hunk 3]
```diff
@@ -69,28 +69,28 @@ func (la LogicalApply) Init(ctx sessionctx.Context, offset int) *LogicalApply {
...
+	p.basePhysicalPlan = newBasePhysicalPlan(ctx, plancodec.TypeSel, &p, selectOffset)
 	p.childrenReqProps = props
 	p.SetStats(stats)
 	return &p
 }
 
 // Init initializes LogicalUnionScan.
-func (p LogicalUnionScan) Init(ctx sessionctx.Context, offset int) *LogicalUnionScan {
-	p.baseLogicalPlan = newBaseLogicalPlan(ctx, plancodec.TypeUnionScan, &p, offset)
+func (p LogicalUnionScan) Init(ctx sessionctx.Context, selectOffset int) *LogicalUnionScan {
```
Comment (qw4990) [inline pkg/planner/core/initialize.go:86]: Should we use `qbOffset`?

**Review by qw4990 — APPROVED** [review]

File: pkg/util/hint/hint_query_block.go (hunk 2 above)
Comment (hawkingrei) [reply pkg/util/hint/hint_query_block.go:299]: Done

File: pkg/planner/core/initialize.go (hunk 3 above)
Comment (hawkingrei) [reply pkg/planner/core/initialize.go:86]: all ```selectOffset``` have been replaced with ```qbOffset```.

File: pkg/planner/core/plan.go (hunk 1 above)
Comment (hawkingrei) [reply pkg/planner/core/plan.go:70]: Done

File: pkg/planner/core/internal/base/plan.go [hunk 4]
```diff
@@ -108,9 +108,9 @@ func (p *Plan) SetTP(tp string) {
 	p.tp = tp
 }
 
-// SelectBlockOffset is to get the select block offset.
-func (p *Plan) SelectBlockOffset() int {
-	return p.blockOffset
+// QBOffset is to get the select block offset.
+func (p *Plan) QBOffset() int {
```
Comment (hi-rustin) [inline pkg/planner/core/internal/base/plan.go:112]: May I ask why we call it `QB`? Why not just call it `QueryBlock`? It is more obvious and less confusing.

**Review by hi-rustin — APPROVED** [review]
Looks good to me. Thanks! 👍

File: pkg/planner/core/internal/base/plan.go (hunk 4 above)
Comment (hi-rustin) [reply pkg/planner/core/internal/base/plan.go]: Got it. Make sense. Thank you. I guess QueryBlockOffset is also short enough. I don't have a strong opinion about it. So feel free to change it.

Comment (hawkingrei) [reply pkg/planner/core/internal/base/plan.go]: You are right. I have fixed it.

**Pull request merged**